Renaming base url setting (#8268)
wagtail/wagtail: docs/reference/settings.rst

.. _this Google Search Central Blog post: https://developers.google.com/search/blog/2010/04/to-slash-or-not-to-slash

ADMIN BASE URL
==============

``WAGTAILADMIN_BASE_URL``
--------------------------

.. code-block:: python

    WAGTAILADMIN_BASE_URL = 'http://example.com/'

> This is the base URL used by the Wagtail admin site. It is typically used for generating URLs to include in emails.This setting was previously named 'BASE_URL'.

Reviewer: ```suggestion
This is the base URL used by the Wagtail admin site. It is typically used for generating URLs to include in notification emails.
```
Remove the previous name here, we will add that below (see other comment), plus a small fix for the usage note.

> If this setting is not present, Wagtail will try to fall back to ``request.site.root_url`` or to the requests host name

Reviewer: ```suggestion
If this setting is not present, Wagtail will try to fall back to ``request.site.root_url`` or to the requests host name.
```
Just a full stop at the end.

> 

Reviewer: We can add a RST syntax that helps for changes
```
.. versionchanged:: 2.15

This setting was previously named ``BASE_URL`` and was undocumented, using ``BASE_URL`` will be removed in a future release.
```

.. _this Google Search Central Blog post: https://developers.google.com/search/blog/2010/04/to-slash-or-not-to-slash

ADMIN BASE URL
==============

``WAGTAILADMIN_BASE_URL``
--------------------------

.. code-block:: python

    WAGTAILADMIN_BASE_URL = 'http://example.com/'

This is the base URL used by the Wagtail admin site. It is typically used for generating URLs to include in notification emails.

> If this setting is not present, Wagtail will try to fall back to ``request.site.root_url`` or to the requests host name.

Reviewer: ```suggestion
If this setting is not present, Wagtail will try to fall back to ``request.site.root_url`` or to the request's host name.
```

> .. versionchanged:: 2.15

Reviewer: sorry the 2.15 was just an example, the next release is 3.0
```suggestion
.. versionchanged:: 3.0
```

Author: oops sorry didn't realize
